xonsh/xoreutils/cat.py

```diff
@@ -44,7 +45,7 @@ def _cat_single_file(opts, fname, stdin, out, err, line_count=1):
     read_size = 0
     file_size = fobj = None
     if fname == "-":
-        f = stdin
+        f = stdin or sys.stdin
```

Comment:
In my case stdin was always `None` when calling `cat` with no arguments. This fixes  #2365, but maybe the error is somewhere else.